yarn-project/circuit-types/src/l2_block.ts

```diff
@@ -432,6 +219,14 @@ export class L2Block {
     return this.#l1BlockNumber;
   }
 
+  /**
+   * Sets the L1 block number that included this block
+   * @param l1BlockNumber - The block number of the L1 block that contains this L2 block.
+   */
+  public setL1BlockNumber(l1BlockNumber: bigint) {
```

Comment:
Unrelated to your work but oh boy I really hate this setter.

I see that it was used in the past in the getBlockFromCallData function when the block number was not serialized in the block buffer.

Could you please nuke it and the `this.#l1BlockNumber` property as well?

Reply:
@sklppy88 It's in global variables of a header.

Reply:
@sklppy88 Oh, I am retarded and read it as l2 block number. Ignore me.

But anyway it feels like a terrible antipattern to have a value on an object which does not get serialized and I think we should just avoid doing that (I think it's a sign of shit code as you `Class.fromBuffer(object.toBuffer())` should always give you the same thing).

But it can absolutely be done in a separate PR as it's not related to the changes you did.

Reply:
Could you create an issue for this and link it here?